src/poetry/console/commands/self/update.py

```diff
@@ -36,192 +30,30 @@ class SelfUpdateCommand(Command):
-
-        pool = Pool()
-        pool.add_repository(PyPiRepository())
-
-        return pool
-
-    def handle(self) -> int:
-        from poetry.core.packages.dependency import Dependency
-        from poetry.core.semver.version import Version
-
-        from poetry.__version__ import __version__
-
-        version = self.argument("version")
-        if not version:
-            version = ">=" + __version__
-
-        repo = self.pool.repositories[0]
-        packages = repo.find_packages(
-            Dependency("poetry", version, allows_prereleases=self.option("preview"))
-        )
-        if not packages:
-            self.line("No release found for the specified version")
-            return 1
-
-        def cmp(x: Package, y: Package) -> int:
-            if x.version == y.version:
-                return 0
-            return int(x.version < y.version or -1)
-
-        packages.sort(key=cmp_to_key(cmp))
-
-        release = None
-        for package in packages:
-            if package.is_prerelease():
-                if self.option("preview"):
-                    release = package
-
-                    break
-
-                continue
-
-            release = package
-
-            break
-
-        if release is None:
-            self.line("No new release found")
-            return 1
-
-        if release.version == Version.parse(__version__):
-            self.line("You are using the latest version")
-            return 0
-
-        self.line(f"Updating <c1>Poetry</c1> to <c2>{release.version}</c2>")
-        self.line("")
-
-        self.update(release)
-
-        self.line("")
-        self.line(
-            f"<c1>Poetry</c1> (<c2>{release.version}</c2>) is installed now. Great!"
-        )
-
-        return 0
-
-    def update(self, release: Package) -> None:
-        from poetry.utils.env import EnvManager
-
-        version = release.version
-
-        env = EnvManager.get_system_env(naive=True)
-
-        # We can't use is_relative_to() since it's only available in Python 3.9+
-        try:
-            env.path.relative_to(self.data_dir)
-        except ValueError:
-            # Poetry was not installed using the recommended installer
-            from poetry.console.exceptions import PoetrySimpleConsoleException
-
-            raise PoetrySimpleConsoleException(
-                "Poetry was not installed with the recommended installer, "
-                "so it cannot be updated automatically."
+    help = """\
+The <c1>self update</c1> command updates Poetry version in its current runtime \
+environment.
+"""
+
+    def _system_project_handle(self) -> int:
+        self.write("<info>Updating Poetry version ...</info>\n\n")
+        application = cast(Application, self.application)
+        add_command: AddCommand = cast(AddCommand, application.find("add"))
+        add_command.set_env(self.env)
+        application._configure_installer(add_command, self._io)
```

Comment:
We should think about renaming `_configure_installer` to `configure_installer` if there is a need for public access to this method.

Reply:
I agree. Fixed this.